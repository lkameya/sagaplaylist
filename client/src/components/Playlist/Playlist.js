import React from 'react';
import PlaylistItem from '../PlaylistItem/PlaylistItem';
import styled from 'styled-components';

const PlaylistContainer = styled.div`
  display: flex;
  overflow-x: scroll;
  white-space: nowrap;
  justify-content: center;
  width: 100vw;
  padding: 2vh;
  padding-top: 5vh;
`;

const Playlist = ({ playlist, currentSongId }) => {
  if (playlist) {
    return (
      <PlaylistContainer>
        {
          playlist.map(song => {
            return (
              <PlaylistItem
                id={song.id}
                key={song.id}
                title={song.name}
                artist={song.artists[0].name}
                albumImg={song.album.images[1].url}
              />
            )
          })
        }
      </PlaylistContainer>
    );
  }

  return null;
}

export default Playlist;
