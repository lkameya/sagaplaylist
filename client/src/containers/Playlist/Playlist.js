import React, { Component } from 'react';
import PlaylistItem from '../../components/PlaylistItem/PlaylistItem';
import styles from './Playlist.module.css';

class Playlist extends Component {
  render() {
    if(this.props.playlist) {
      return (
        <div className={styles.playlist} >
          {this.props.playlist.map(song => {
            return (
              <PlaylistItem
                id={song.id}
                key={song.id}
                title={song.name}
                artist={song.artists[0].name}
                albumImg={song.album.images[1].url}
                />
            )
          })}
        </div>
      );
    } else {
      return null;
    }

  }
}

export default Playlist;
