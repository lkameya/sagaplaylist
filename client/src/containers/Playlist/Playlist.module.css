.playlist {
  display: flex;
  overflow-x: scroll;
  white-space: nowrap;
  justify-content: center;
  width: 100vw;
  position: absolute;
  bottom: 2vh;
  padding: 2vh;
}