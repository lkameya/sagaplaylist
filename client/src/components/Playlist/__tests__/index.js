import React from 'react';
import Playlist from '../Playlist';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Playlist', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Playlist>
      </Playlist>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
