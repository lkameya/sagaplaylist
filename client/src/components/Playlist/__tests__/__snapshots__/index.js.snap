// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist renders correctly 1`] = `null`;
